### PR TITLE
Fix bug 1567394 - Allow access to issue tracker

### PR DESCRIPTION
**Pull request by katherine95**
#### What does this PR do?
This pull request fixes the issue [1567394](https://bugzilla.mozilla.org/show_bug.cgi?id=1567394). It allows a user to access the issue tracker from the Pontoon app and also give feedback via [pontoon's](https://discourse.mozilla.org/c/pontoon) feedback page.

#### Type of change:
- [x] Bug fix (non-breaking change which fixed an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update.

#### How Has This Been Tested?
- Clone this repository 
- Git checkout to `katherine95:access-issue-tracker`branch.
- Follow the setup guidelines to set up pontoon locally
- Run `make run`
- Login to Pontoon
- Navigate to your profile and click on it.
- Click on `Hack it on Github`  and `Give feedback`.
- Both should redirect you to Github and pontoon's feedback page respectively.

#### Checklist:
- [x] On click on all the stated links on the user's profile should open the required pages on a new tab
- [x] The added links; Hack it on Github and Give feedback should redirect to Github and pontoon's feedback page respectively.

#### Screenshots (if appropriate):
image

### CODE REVIEW

File: pontoon/base/static/css/fontawesome-all.css [hunk 1]
```diff
@@ -13,7 +13,11 @@
   font-style: normal;
   font-variant: normal;
   text-rendering: auto;
-  line-height: 1; }
+  line-height: 1;}
+
+.fab-gh {
+  width: 1.8em;
+}
```
Comment (mathjazz) [inline pontoon/base/static/css/fontawesome-all.css]: This is a CSS template by Font Awesome, so we don't edit it. Let's revert this change.

File: pontoon/base/templates/header.html [hunk 2]
```diff
@@ -39,8 +39,10 @@
             <ul>
               {{ Profile.top_menu() }}
 
-              <li><a href="{{ url('pontoon.terms') }}"><i class="fa fa-gavel fa-fw"></i>Terms of Use</a></li>
-              <li><a href="https://mozilla-l10n.github.io/localizer-documentation/tools/pontoon/"><i class="fa fa-life-ring fa-fw"></i>Help</a></li>
+              <li><a href="{{ url('pontoon.terms') }}" rel="noopener noreferrer" target="_blank"><i class="fa fa-gavel fa-fw"></i>Terms of Use</a></li>
+              <li><a href="https://github.com/mozilla/pontoon/" rel="noopener noreferrer" target="_blank"><i class="fab fa-github fab-gh"></i>Hack it on GitHub</a></li>
```
Comment (mathjazz) [inline pontoon/base/templates/header.html]: Instead of adding a custom `fab-gh` class, you should add `fa-fw`, which other icons use to set the [fixed width](https://fontawesome.com/v4.7.0/examples/#fixed-width).

Then, you'll also need to set the same margin as other icons use, which is done in [`style.css`](https://github.com/mozilla/pontoon/blob/1575bfbbd28e55fe72fabffa668cfa55ed04511c/pontoon/base/static/css/style.css#L951). You'll need to extend the rule to also include `#profile .menu li i.fab`:
```css
#profile .menu li i.fa,
#profile .menu li i.fab {
  ...
}
```

File: pontoon/base/static/css/fontawesome-all.css (hunk 1 above)
Comment (katherine95) [reply pontoon/base/static/css/fontawesome-all.css]: Noted working on this.

File: pontoon/base/templates/header.html (hunk 2 above)
Comment (katherine95) [reply pontoon/base/templates/header.html]: Thanks @mathjazz  for feedback, I am currently implementing it. Is there any problem if I git ammend --no-edit and force push?

Comment (mathjazz) [reply pontoon/base/templates/header.html]: No need to amend, simply commit changes separately - that way it's easier to review between review rounds.

Before pushing upstream, I'll merge all commits into one.

Comment (katherine95) [reply pontoon/base/templates/header.html]: @mathjazz having the fixed width, `fa-fw` for icons doesn't fix the width of the Github icon. 
I have also tried extending the same margin as other icons use include `#profile .menu li i.fab` and still it doesn't fix it. 
Screenshot 2019-10-08 at 11 15 08

When I try to set a custom `fab-gh` class in `style.css` for the github icon it works
Screenshot 2019-10-08 at 11 19 47

Comment (mathjazz) [reply pontoon/base/templates/header.html]: @katherine95 Maybe there's some caching issue? I've tried the proposed change locally and it works, which is expected - because all the icons should use the same styling.

You can try to run `make build` and then `make run` again to make sure all files are updated.

Comment (katherine95) [reply pontoon/base/templates/header.html]: Yes, it has now worked after running `make build`. @mathjazz I would like to test for the changes I am making in `frontend/src/core/user/components/UserMenu.js` component. How can I access/reproduce the User interface powered by that component?

Comment (mathjazz) [reply pontoon/base/templates/header.html]: This files power our new Translate UI, which we're now rolling out to users.

There's one command you need to run within `make shell` to turn on the new UI:
https://github.com/mozilla/pontoon/blob/master/frontend/README.md#this-feature-is-behind-a-switch

Comment (katherine95) [reply pontoon/base/templates/header.html]: I have followed the steps in the README file. Where can I access the new Translate UI? @mathjazz

Comment (mathjazz) [reply pontoon/base/templates/header.html]: Just open any translate view, e.g. http://localhost:8000/projects/tutorial/playground/.

The changes compared to the old UI are very subtle, the most noticeable one being the new navigation bar. https://ffp4g1ylyit3jdyti1hqcvtb-wpengine.netdna-ssl.com/l10n/files/2019/09/new.png

Comment (katherine95) [reply pontoon/base/templates/header.html]: Got it thanks

Comment (katherine95) [reply pontoon/base/templates/header.html]: @mathjazz I am facing some blocker, after activating the new UI using `./manage.py waffle_flag translate_next --everyone --create` accessing any of the tranlate languages doesn't load the page, for example `http://localhost:8000/translate/af/tutorial/all-resources/?status=translated` can't load, this is a sample screenshot.
Screenshot 2019-10-14 at 13 30 32

Comment (mathjazz) [reply pontoon/base/templates/header.html]: You should see an error in the DevTools Console if you scroll up. Could you paste the full error here?

Comment (mathjazz) [reply pontoon/base/templates/header.html]: It turns out we have a bug!
https://bugzilla.mozilla.org/show_bug.cgi?id=1588565

Comment (katherine95) [reply pontoon/base/templates/header.html]: I see. I will just push my current changes as of now. Is that okay? @mathjazz

File: pontoon/base/static/css/fontawesome-all.css [hunk 3]
```diff
@@ -13,7 +13,7 @@
   font-style: normal;
   font-variant: normal;
   text-rendering: auto;
-  line-height: 1; }
+  line-height: 1;}
```
Comment (mathjazz) [inline pontoon/base/static/css/fontawesome-all.css]: Please revert this change.

File: pontoon/base/static/css/style.css [hunk 4]
```diff
@@ -948,7 +948,8 @@ header .right .select .menu {
   line-height: 22px;
 }
 
-#profile .menu li i.fa {
+#profile .menu li i.fa,
+#profile .menu li i.fab {
```
Comment (mathjazz) [inline pontoon/base/static/css/style.css]: You'll need to make a similar change here to properly position the GitHub link on the translate page:
https://github.com/mozilla/pontoon/blob/8f1663fd3ea53c9f29b3d2f9f90d92b56c62b5e2/frontend/src/core/user/components/UserMenu.css#L87

You'll need to add `.user-menu .menu li i.fab` to the rule.

File: frontend/src/core/user/components/UserMenu.js [hunk 5]
```diff
@@ -173,6 +173,32 @@ export class UserMenuBase extends React.Component<Props, State> {
...
+                </li>
+
+                <li>
+                    <Localized
+                        id="user-UserMenu--feedback"
+                        glyph={
+                            <i className="fa fa-comment-dots fa-fw"></i>
+                        }
+                    >
+                        <a href="https://discourse.mozilla.org/c/pontoon">
```
Comment (mathjazz) [inline frontend/src/core/user/components/UserMenu.js]: Nice work!

You'll also need to add `rel="noopener noreferrer" target="_blank"` to the `<a>` elements, e.g.:
```html
<a
    href="https://github.com/mozilla/pontoon/"
    rel="noopener noreferrer"
    target="_blank"
>
    { '<glyph></glyph>Hack it on GitHub' }
</a>
```

Please do that for Terms, GitHub, Feedback and Help links.

And while you're at it, please remove the Top Contributors and the Machinery links (as well as their translations from the .ftl files).

Comment (katherine95) [reply frontend/src/core/user/components/UserMenu.js]: Done. @mathjazz

File: frontend/src/core/user/components/UserMenu.js [hunk 6]
```diff
@@ -136,39 +136,45 @@ export class UserMenuBase extends React.Component<Props, State> {
...
                         glyph={
-                            <i className="fa fa-trophy fa-fw"></i>
+                            <i className="fa fa-gavel fa-fw"></i>
                         }
                     >
-                        <a href="/contributors/">
-                            { '<glyph></glyph>Top Contributors' }
+                        <a href="/terms/"
+                        rel="noopener noreferrer"
+                        target="_blank">
```
Comment (mathjazz) [inline frontend/src/core/user/components/UserMenu.js]: Nit: We try to indent attributes like this:
```html
<a
    href="/terms/"
    rel="noopener noreferrer"
    target="_blank"
>
```
Please fix that for all four links:

File: frontend/public/static/locale/en-US/translate.ftl [hunk 7]
```diff
@@ -475,9 +475,9 @@ user-UserMenu--download-tm = <glyph></glyph>Download Translation Memory
 user-UserMenu--download-translations = <glyph></glyph>Download Translations
 user-UserMenu--upload-translations = <glyph></glyph>Upload Translations
 
-user-UserMenu--top-contributors = <glyph></glyph>Top Contributors
-user-UserMenu--machinery = <glyph></glyph>Machinery
```
Comment (mathjazz) [inline frontend/public/static/locale/en-US/translate.ftl]: Sorry for not noticing this before - could you please remove these two entries also from the fr file?
https://github.com/mozilla/pontoon/blob/2fb5ea62fb474ef219c926ae41b91e6c64be5bd7/frontend/public/static/locale/fr/translate.ftl#L302

Comment (katherine95) [reply frontend/public/static/locale/en-US/translate.ftl]: How about for Hack it on Github and Give feedback? They are missing in that file. Should I add them? @mathjazz

Comment (mathjazz) [reply frontend/public/static/locale/en-US/translate.ftl]: That's OK - the fr and es files are not fully localized yet. We have to fix a few more [localizability bugs](https://wiki.mozilla.org/L10n:Pontoon#Translate.Next_Followup) and then we'll start localizing Pontoon itself on pontoon.mozilla.org.

**Comment by katherine95** [comment]
> Well done, @katherine95!
> 
> Thank you for your contribution and welcome to Pontoon :)

Thank you @mathjazz. It has been a great learning experience.
